pkg/nodejs/pnpm/parse.go

```diff
@@ -0,0 +1,136 @@
+package pnpm
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"golang.org/x/xerrors"
+	"gopkg.in/yaml.v3"
+
+	dio "github.com/aquasecurity/go-dep-parser/pkg/io"
+
+	"github.com/aquasecurity/go-dep-parser/pkg/types"
+)
+
+type PackageResolution struct {
+	Integrity string `yaml:"integrity"`
+}
+
+type PackageInfo struct {
+	Resolution           PackageResolution `yaml:"resolution"`
+	Engines              map[string]string `yaml:"engines,omitempty"`
+	Specifiers           map[string]string `yaml:"specifiers,omitempty"`
+	Dependencies         map[string]string `yaml:"dependencies,omitempty"`
+	OptionalDependencies map[string]string `yaml:"optionalDependencies,omitempty"`
+	DevDependencies      map[string]string `yaml:"devDependencies,omitempty"`
+	IsDev                bool              `yaml:"dev,omitempty"`
+	IsOptional           bool              `yaml:"optional,omitempty"`
+}
+
+type LockFile struct {
+	LockfileVersion      int8                   `yaml:"lockfileVersion"`
+	Importers            map[string]PackageInfo `yaml:"importers,omitempty"`
+	Specifiers           map[string]string      `yaml:"specifiers,omitempty"`
+	Dependencies         map[string]string      `yaml:"dependencies,omitempty"`
+	OptionalDependencies map[string]string      `yaml:"optionalDependencies,omitempty"`
+	DevDependencies      map[string]string      `yaml:"devDependencies,omitempty"`
+	Packages             map[string]PackageInfo `yaml:"packages,omitempty"`
+}
+
+type Parser struct{}
+
+func NewParser() types.Parser {
+	return &Parser{}
+}
+
+func (p *Parser) ID(name, version string) string {
+	return fmt.Sprintf("%s@%s", name, version)
+}
+
+func (p *Parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency, error) {
+	var lockFile LockFile
+	decoder := yaml.NewDecoder(r)
+	err := decoder.Decode(&lockFile)
+	if err != nil {
+		return nil, nil, xerrors.Errorf("decode error: %w", err)
+	}
+
+	libs, deps := p.parse(&lockFile)
+
+	return uniqueLibs(libs), uniqueDeps(deps), nil
```

Comment:
Do we need deduplication here? IIUC, we iterate `packages` that are unique.

Reply:
I have done some checking and you are correct! I have removed the unnecessary deduplication

let me know if is ok now :)